app/models/job_run.rb

```diff
@@ -51,8 +51,13 @@ def mail_subject
     "[#{batch_context.project_name}] Your #{job_type.humanize} job completed"
   end
 
+  # the states that indicate this job is either not staretd or is currently running
+  def in_progress?
+    (waiting? || running?)
+  end
+
   def send_notification
-    return unless output_location
+    return if in_progress?
 
```

Comment:
guard condition now uses the state machine instead of the database field (which is no longer updated for pre-assembly jobs)